unchained-commons/src/main/java/unchained/commons/utils/ReflectionUtils.java

```diff
@@ -20,6 +22,310 @@ public static Type resolveCommonAncestor(List<Type> types) {
+                    populateSuperTypeArgs(((Class<?>) rawType).getGenericInterfaces(), map, depthFirst);
+                }
+                if (depthFirst) {
+                    populateTypeArgs(parameterizedType, map, depthFirst);
+                }
+            } else if (type instanceof Class) {
+                populateSuperTypeArgs(((Class<?>) type).getGenericInterfaces(), map, depthFirst);
+            }
+        }
+    }
+
+    private static void populateTypeArgs(ParameterizedType type, Map<TypeVariable<?>, Type> map, boolean depthFirst) {
+        if (type.getRawType() instanceof Class) {
+            TypeVariable<?>[] typeVariables = ((Class<?>) type.getRawType()).getTypeParameters();
+            Type[] typeArguments = type.getActualTypeArguments();
+            if (type.getOwnerType() != null) {
+                Type owner = type.getOwnerType();
+                if (owner instanceof ParameterizedType) {
+                    populateTypeArgs((ParameterizedType) owner, map, depthFirst);
+                }
+            }
+            for (int i = 0; i < typeArguments.length; i++) {
+                TypeVariable<?> variable = typeVariables[i];
+                Type typeArgument = typeArguments[i];
+                if (typeArgument instanceof Class) {
+                    map.put(variable, typeArgument);
+                } else if (typeArgument instanceof GenericArrayType) {
+                    map.put(variable, typeArgument);
+                } else if (typeArgument instanceof ParameterizedType) {
+                    map.put(variable, typeArgument);
+                } else if (typeArgument instanceof TypeVariable) {
+                    TypeVariable<?> typeVariableArgument = (TypeVariable<?>) typeArgument;
+                    if (depthFirst) {
+                        Type existingType = map.get(variable);
+                        if (existingType != null) {
+                            map.put(typeVariableArgument, existingType);
+                            continue;
+                        }
+                    }
+                    Type resolvedType = map.get(typeVariableArgument);
+                    if (resolvedType == null) {
+                        resolvedType = resolveBound(typeVariableArgument);
+                    }
+                    map.put(variable, resolvedType);
+                }
+            }
+        }
+    }
+
+    private static Type resolveBound(TypeVariable<?> typeVariable) {
+        Type[] bounds = typeVariable.getBounds();
+        if (bounds.length == 0) {
+            return Unknown.class;
+        }
+        Type bound = bounds[0];
+        if (bound instanceof TypeVariable) {
+            bound = resolveBound((TypeVariable<?>) bound);
+        }
+        return bound == Object.class ? Unknown.class : bound;
+    }
+
+    private static void populateLambdaArgs(Class<?> functionalInterface, final Class<?> lambdaType, Map<TypeVariable<?>, Type> map) {
```

Comment:
This method is too complex to review or maintain. Could you break it into smaller pieces?